File: test/easyswift/ReverseStringTest.swift
import Testing

@testable import easy

@Test(arguments: [
  (["h", "e", "l", "l", "o"], ["o", "l", "l", "e", "h"]),
  (["a", "b", "c", "d"], ["d", "c", "b", "a"]),
  (["z"], ["z"]),
  ([], []),
])
func reverseStringTest(input: [Character], outcome: [Character]) {
  var mutableInput = input
  reverseString(&mutableInput)
  #expect(mutableInput == outcome)
}

File: test/easy/ReverseStringTest.php

<?php

use Src\Easy\ReverseString;

describe('TestValue', function () {

    test('Testing', function ($candidate, $expected) {
        $instance = new ReverseString;
        $instance->solve($candidate);
        expect($candidate)->toBe($expected);

    })->with([
        [['h', 'e', 'l', 'l', 'o'], ['o', 'l', 'l', 'e', 'h']],
        [['a', 'b', 'c', 'd'], ['d', 'c', 'b', 'a']],
        [['z'], ['z']],
        [[], []],
    ]);

});
